### PR TITLE
[WFLY-18511] thread-racing Quickstart CY2023Q3 enhancements

### PR DESCRIPTION
**Pull request by emmartins**
Issue: https://issues.redhat.com/browse/WFLY-18511

### CODE REVIEW

File: thread-racing/README.adoc [hunk 1]
```diff
@@ -60,13 +60,13 @@ The application will be running at the following URL http://localhost:8080/{arti
...
+include::../shared-doc/run-integration-tests-with-server-distribution.adoc[leveloffset=+2]
 // Undeploy the Quickstart
 include::../shared-doc/undeploy-the-quickstart.adoc[leveloffset=+2]
 
 // Build and run sections for other environments/builds
 ifndef::ProductRelease,EAPXPRelease[]
 include::../shared-doc/build-and-run-the-quickstart-with-provisioned-server.adoc[leveloffset=+1]
 endif::[]
-include::../shared-doc/build-and-run-the-quickstart-with-openshift.adoc[leveloffset=+1]
+include::../shared-doc/build-and-run-the-quickstart-with-openshift.adoc[leveloffset=+1]
```
Comment (marcosgopen) [inline thread-racing/README.adoc:72]: @emmartins I tested the application with helm on OpenShift but I found very difficult to use this adoc file.
In particular [this line](https://github.com/wildfly/quickstart/blob/main/shared-doc/helm-deploy-project.adoc?plain=1#L10) explains how to install the chart: "helm install {artifactId} -f charts/helm.yaml {helmChartName} --wait --timeout=10m0s".
IMO it is not obvious that I have to replace {helmChartName} with wildfly/wildfly. WDYT?

Comment (emmartins) [reply thread-racing/README.adoc:72]: This is due to wfly to eap rebranding needs, it uses an attribute configured at /shared-docs/attributes.adoc , the best way to work with docs is to keep rebuilding it when doing changes (please use -Pdocs or -Drelease for the doc rebuild)

File: thread-racing/pom.xml [hunk 2]
```diff
@@ -345,33 +300,22 @@
...
             <build>
                 <plugins>
                     <plugin>
                         <groupId>org.apache.maven.plugins</groupId>
                         <artifactId>maven-failsafe-plugin</artifactId>
-                        <version>${version.failsafe.plugin}</version>
                         <configuration>
                             <includes>
+                                <include>**/BasicRuntimeIT</include>
                                 <include>**/RemoteThreadRacingIT</include>
```
[the remark this answers is not in the document]
Comment (emmartins) [reply thread-racing/pom.xml:312]: Yes, that's what I did in this PR.

File: thread-racing/README.adoc (hunk 1 above)
Comment (marcosgopen) [reply thread-racing/README.adoc:72]: I see! Thanks!

**Pull request merged**